### ISSUE TITLE
Customer VAT ID not validated when an address is saved / updated in customer dashboard

### ISSUE THREAD

**Issue by jakwinkler** [magento/magento2#30720]
### Preconditions (*)

1. Any Magento 2 version
2. Setup proper customer groups for EU VAT ID validation
3. Make sure TAX / VAT field is visible in front so customer can edit it
4. Configure Shipping Origin and store owner data as a valid EU entity with valid VATEU number

### Steps to reproduce (*)

1. Create an account Go to customer dashboard
2. Add new address and fill TAX FIELD
3. save new address

### Expected result (*)

1. Customer is assigned to customer group (for example B2B customer which are not charged with VAT)

### Actual result (*)

1. Vat ID validation error:
`Your Tax ID cannot be validated. You will be charged tax. If you believe this is an error, please contact us at`
2. Customer group is not changed

Class `\Magento\Customer\Observer\AfterAddressSaveObserver` is bugged, vat validation does not use requester ID when it validates customer address VAT ID

```
$result = $this->_customerVat->checkVatNumber(
                    $customerAddress->getCountryId(),
                    $customerAddress->getVatId()
                );
```

however in this class `\Magento\Quote\Observer\Frontend\Quote\Address\VatValidator` , Tax Validation looks like this:

```
$validationResult = $this->customerVat->checkVatNumber(
                $customerCountryCode,
                $customerVatNumber,
                $merchantVatNumber !== '' ? $merchantCountryCode : '',
                $merchantVatNumber
            );
```

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by peterjaap** [comment]
Magento 2 uses the VIES validation service, which *requires* a Requester Member State & number (in Magento's terms a merchant VAT number and merchant country code). See https://ec.europa.eu/taxation_customs/vies/vatRequest.html.

**Comment by qsolutions-pl** [comment]
has anyone confirmed the issue?

**Comment by dverkade** [comment]
Posting a comment. Issues should not be automatically closed.

**Comment by engcom-Alfa** [comment]
Hi @qsolutions-pl 
Need more info , in fact Tried to reproduce it on Magento ver. dev-2.4-develop but i could not get the error provided for EU. Hence request to provide more test data used for this scenario like Zip code, Country and TAX VAT Number.

**Comment by engcom-Lima** [comment]
Hi @qsolutions-pl ,

Waited since 2 week about the updates, Hence closing this ticket as of now. If you have any issue or query then you can raise the new ticket or reopen.

Thanks for understanding!

**Comment by hostep** [comment]
This issue seems to be closed for the incorrect reasons.

The offending code referred to in the opening comment is still not passing on the merchant VAT data on the `2.4-develop` branch: https://github.com/magento/magento2/blob/ed2ffad8f227e253c0aa50393f40478b02310b65/app/code/Magento/Customer/Observer/AfterAddressSaveObserver.php#L152-L155

I'll try to provide more info about how to reproduce this later when I find some time.

**Comment by hostep** [comment]
Hmm, I just tried to reproduce and it turns out that VIES validation does work without passing the requester member state country and number. Even on [their own form](https://ec.europa.eu/taxation_customs/vies/vatRequest.html) those last two fields can remain empty and it still works.

So the issue is not valid (or at least not until VIES makes this a hard requirement one day in the future).

@qsolutions-pl: do you still have this problem or has it gone away?
